core/src/main/java/edu/ucr/cs/riple/core/util/Utility.java

```diff
@@ -307,14 +306,12 @@ public static ProgressBar createProgressBar(String taskName, int steps) {
   }
 
   public static void writeLog(Config config) {
-    File file = config.globalDir.resolve("log.txt").toFile();
-    try (FileOutputStream fos = new FileOutputStream(file)) {
-      BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(fos));
+    Path path = config.globalDir.resolve("log.txt");
+    try (BufferedWriter bw = Files.newBufferedWriter(path, Charset.defaultCharset()); ) {
```

Comment:
Unless we are doing it for some performance reasons, you could use the `Files` api too. `Files.write(...)`

Reply:
No it is not for performance, rewrote it with `Files` api. 48d4a6470593935cbb2025c6fd13848f21344672